WearOAuth/oauth-device-grant/src/main/java/com/example/android/wearable/oauth/devicegrant/AuthDeviceGrantViewModel.kt

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.wearable.oauth.devicegrant
+
+import android.content.Intent
+import android.net.Uri
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.wear.remote.interactions.RemoteIntentHelper
+import com.example.android.wearable.oauth.util.doGetRequest
+import com.example.android.wearable.oauth.util.doPostRequest
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+private const val TAG = "AuthDeviceGrantViewModel"
+
+// TODO Add your client id & secret here (for dev purposes only).
+private const val CLIENT_ID = ""
+private const val CLIENT_SECRET = ""
+
+/**
+ * The viewModel that implements the OAuth flow. The method [startAuthFlow] implements the
+ * different steps of the flow. It first retrieves the URL that should be opened on the paired
+ * device, then polls for the access token, and uses it to retrieve the user's name.
+ */
+class AuthDeviceGrantViewModel(private val remoteIntentHelper: RemoteIntentHelper) : ViewModel() {
+    // Status to show on the Wear OS display
+    val status: MutableLiveData<String> by lazy { MutableLiveData<String>() }
+    private fun showStatus(statusString: String) = status.postValue(statusString)
+
+    fun startAuthFlow() {
+        viewModelScope.launch(Dispatchers.IO) {
+            // Step 1: Retrieve the verification URI
+            showStatus("Starting authorization... Switch to your phone to authenticate.")
```

Comment:
Nit: I think we should be good stewards and avoid having hardcoded displayed text that isn't localizable.

Reply:
I agree with your point. However, moving to localizable strings complicates this sample too much in my opinion. As per [this blog](https://medium.com/androiddevelopers/locale-changes-and-the-androidviewmodel-antipattern-84eb677660d9), we shouldn't access string resources in the view model. As some of our strings are hard-coded, and some contain variables, this would not be trivial to pass from the view model to the activity. What do you think?

Reply:
Ah true, that `ViewModel` gotcha and the parameterized string resources make that a bit more complicated...

I still think it'd be good to add the work necessary to the sample to handle that, even if it does make the sample more complicated overall. To me it feels a bit disingenuous to leave a 
"TODO: Convert to resources, but watch out for [this](https://medium.com/androiddevelopers/locale-changes-and-the-androidviewmodel-antipattern-84eb677660d9)" undone.

Either way, don't want to hold up this PR for that!

Reply:
You're right, and thanks for holding me accountable. I updated the sample to allow for a hard-coded and a dynamic text view. The hard-coded one is now based on resources.